frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1651,9 +1658,27 @@ impl<T: Config> ElectionProvider for Pallet<T> {
 				Ok(supports)
 			},
 			Err(why) => {
-				log!(error, "Entering emergency mode: {:?}", why);
-				<CurrentPhase<T>>::put(Phase::Emergency);
-				Err(why)
+				// if `elect()` was called during `Phase::Off` or too early since the beginning of
+				// the signed phase, do not enter in emergency mode since there was no enough time
+				// to reach a solution.
+				match CurrentPhase::<T>::get() {
+					Phase::Off => {
+						Self::deposit_event(Event::EmergencyPhaseThrottled);
+						Err(why)
+					},
+					Phase::Signed(started_at)
+						if (started_at + T::MinSignedPhaseDuration::get()) >
```

Comment:
We could just use `SignedPhase` itself instead of `MinSignedPhaseDuration `. The reasoning being, for election to fail and go into emergency phase, we should at least wait for the full duration of signed phase.

Reply:
I think that could be a good way fwd, yes. I'm wondering though if having more configuration options here would be helpful? E.g, with `T::MinSignedPhaseDuration` (btw, I will rename this, it's confusing), we could:

1. set `T::MinSignedPhaseDuration` to 0 --> current behaviour.
2. set `T::MinSignedPhaseDuration` to `SignedPhase` (duration in blocks of the signed phase) --> it mimics the behaviour you described above.  
3. set `T::MinSignedPhaseDuration` to anything between [0, `T::SignedPhase`]

Btw, this makes me realise that if `T::MinSignedPhaseDuration` > `SignedPhase` we may never enter in emergency mode. That's no good and could be a relatively silent bug. So I suggest to change the `T::MinSignedPhaseDuration` to  a perbill of the signed duration phase: `type MinSignedBlocksBeforeEmergency: Get<Perbill>`

Reply:
The [latest commit](https://github.com/paritytech/substrate/pull/13040/commits/7ffcd18e8e99e72c77a095693c9e28a3c9f8d45c) changes the name of `T:: MinSignedPhaseDuration` to `T:: MinSignedBlocksBeforeEmergency` and, more importantly, it's a perbill (representing the % of blocks of `SignedPhase` that we expect an election result to be ready).